features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/timeline/TimelineItemIndexerTest.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.timeline
+
+import com.google.common.truth.Truth.assertThat
+import io.element.android.features.messages.impl.timeline.model.TimelineItem
+import io.element.android.features.messages.impl.timeline.model.virtual.TimelineItemReadMarkerModel
+import io.element.android.libraries.matrix.api.core.EventId
+import io.element.android.libraries.matrix.test.AN_EVENT_ID
+import org.junit.Test
+
+class TimelineItemIndexerTest {
+    @Test
+    fun `test TimelineItemIndexer`() {
+        val eventIds = mutableListOf<EventId>()
+        val data = listOf(
+            aTimelineItemEvent().also { eventIds.add(it.eventId!!) },
```

Comment:
Maybe make sure all ids used here don't match? They're random, so there's not much of a chance of it happening, so up to you.

Reply:
Yes, I guess it's fine. I preferred to keep the randomness and store ids in `eventIds`, another solution would have been to create `AN_EVENT_ID_3` and `AN_EVENT_ID_4`. Let's revisit later if it's worth it.